diff --git a/components/tinyusb/CMakeLists.txt b/components/tinyusb/CMakeLists.txt
index e7c1b7edbc..07c8245c3f 100644
--- a/components/tinyusb/CMakeLists.txt
+++ b/components/tinyusb/CMakeLists.txt
@@ -68,6 +68,13 @@ if(CONFIG_TINYUSB)
             "additions/src/vfs_tinyusb.c"
             )
     endif() # CONFIG_TINYUSB_CDC_ENABLED
+
+    if(CONFIG_TINYUSB_HID_ENABLED)
+        list(APPEND srcs
+            "additions/src/tusb_hid_gamepad.c"
+            )
+    endif() # CONFIG_TINYUSB_CDC_ENABLED
+
 endif() # CONFIG_TINYUSB
 
 idf_component_register(SRCS ${srcs}
diff --git a/components/tinyusb/Kconfig b/components/tinyusb/Kconfig
index 9e28a67713..fa6d92d4de 100644
--- a/components/tinyusb/Kconfig
+++ b/components/tinyusb/Kconfig
@@ -150,6 +150,21 @@ menu "TinyUSB Stack"
                 help
                     CDC FIFO size of TX channel.
         endmenu # "Communication Device Class"
+
+        menu "Human Interface Device (HID)"
+            config TINYUSB_HID_ENABLED
+                bool "Enable TinyUSB HID feature"
+                default y
+                help
+                    Enable TinyUSB HID feature.
+
+            config TINYUSB_HID_BUFSIZE
+                depends on TINYUSB_HID_ENABLED
+                int "HID buffer size"
+                default 255
+                help
+                    HID FIFO size, in bytes.
+        endmenu # "Human Interface Device"
     endif # TINYUSB
 
 endmenu # "TinyUSB Stack"
diff --git a/components/tinyusb/additions/include/tusb_hid_gamepad.h b/components/tinyusb/additions/include/tusb_hid_gamepad.h
new file mode 100644
index 0000000000..06088ee6dc
--- /dev/null
+++ b/components/tinyusb/additions/include/tusb_hid_gamepad.h
@@ -0,0 +1,8 @@
+#ifndef _TUSB_HID_GAMEPAD_H_
+#define _TUSB_HID_GAMEPAD_H_
+
+#include "class/hid/hid_device.h"
+
+void tud_gamepad_report(hid_gamepad_report_t * report);
+
+#endif /* _TUSB_HID_GAMEPAD_H_ */
\ No newline at end of file
diff --git a/components/tinyusb/additions/include_private/descriptors_control.h b/components/tinyusb/additions/include_private/descriptors_control.h
index 461ffa85e9..7c2133e136 100644
--- a/components/tinyusb/additions/include_private/descriptors_control.h
+++ b/components/tinyusb/additions/include_private/descriptors_control.h
@@ -33,7 +33,8 @@ extern "C" {
 #if CFG_TUD_HID
 enum {
     REPORT_ID_KEYBOARD = 1,
-    REPORT_ID_MOUSE
+    REPORT_ID_MOUSE,
+    REPORT_ID_GAMEPAD
 };
 #endif
 
diff --git a/components/tinyusb/additions/src/descriptors_control.c b/components/tinyusb/additions/src/descriptors_control.c
index d7fbf0bda7..9f5d120f9e 100644
--- a/components/tinyusb/additions/src/descriptors_control.c
+++ b/components/tinyusb/additions/src/descriptors_control.c
@@ -22,8 +22,9 @@ static char *s_str_descriptor[USB_STRING_DESCRIPTOR_ARRAY_SIZE];
 
 #if CFG_TUD_HID //HID Report Descriptor
 uint8_t const desc_hid_report[] = {
-    TUD_HID_REPORT_DESC_KEYBOARD(HID_REPORT_ID(REPORT_ID_KEYBOARD), ),
-    TUD_HID_REPORT_DESC_MOUSE(HID_REPORT_ID(REPORT_ID_MOUSE), )
+    TUD_HID_REPORT_DESC_GAMEPAD(HID_REPORT_ID(REPORT_ID_GAMEPAD) ),
+    // TUD_HID_REPORT_DESC_KEYBOARD(HID_REPORT_ID(REPORT_ID_KEYBOARD), ),
+    // TUD_HID_REPORT_DESC_MOUSE(HID_REPORT_ID(REPORT_ID_MOUSE), )
 };
 #endif
 
@@ -120,7 +121,7 @@ uint16_t const *tud_descriptor_string_cb(uint8_t index, uint16_t langid)
  * @return uint8_t const*
  */
 #if CFG_TUD_HID
-uint8_t const *tud_hid_descriptor_report_cb(void)
+uint8_t const *tud_hid_descriptor_report_cb(uint8_t itf)
 {
     return desc_hid_report;
 }
diff --git a/components/tinyusb/additions/src/tusb_hid_gamepad.c b/components/tinyusb/additions/src/tusb_hid_gamepad.c
new file mode 100644
index 0000000000..3ac164349b
--- /dev/null
+++ b/components/tinyusb/additions/src/tusb_hid_gamepad.c
@@ -0,0 +1,7 @@
+#include "tusb_hid_gamepad.h"
+#include "descriptors_control.h"
+
+void tud_gamepad_report(hid_gamepad_report_t * report)
+{
+    tud_hid_report(REPORT_ID_GAMEPAD, report, sizeof(hid_gamepad_report_t));
+}
\ No newline at end of file
diff --git a/components/tinyusb/sdkconfig.rename b/components/tinyusb/sdkconfig.rename
index 6aea8579ac..e32c20ab4d 100644
--- a/components/tinyusb/sdkconfig.rename
+++ b/components/tinyusb/sdkconfig.rename
@@ -19,4 +19,6 @@ CONFIG_USB_MSC_BUFSIZE                  CONFIG_TINYUSB_MSC_BUFSIZE
 CONFIG_USB_CDC_ENABLED                  CONFIG_TINYUSB_CDC_ENABLED
 CONFIG_USB_CDC_RX_BUFSIZE               CONFIG_TINYUSB_CDC_RX_BUFSIZE
 CONFIG_USB_CDC_TX_BUFSIZE               CONFIG_TINYUSB_CDC_TX_BUFSIZE
+CONFIG_USB_HID_ENABLED                  CONFIG_TINYUSB_HID_ENABLED
+CONFIG_USB_HID_BUFSIZE                  CONFIG_TINYUSB_HID_BUFSIZE
 CONFIG_USB_DEBUG_LEVEL                  CONFIG_TINYUSB_DEBUG_LEVEL
